test/extensions/filters/listener/proxy_protocol/proxy_protocol_test.cc

```diff
@@ -1035,6 +1073,58 @@ TEST_P(ProxyProtocolTest, PartialRead) {
   disconnect();
 }
 
+TEST_P(ProxyProtocolTest, PartialV1ReadWithAllowNoProxyProtocol) {
```

Comment:
Also worth testing a case that we have correct v1 or v2 header signature, but invalid data in the rest of header, then ensure we disconnect the connection, and not let the connection passthrough.

Reply:
addressed in https://github.com/envoyproxy/envoy/pull/18951/commits/6031151f9ff7a056fbb2adb874b2ab2f327d7697